### PR TITLE
1292 persist charge pricelists

### PR DESCRIPTION
**Pull request by x-platformcoder**
## Description
Persisting charge prices from messages with business reasoncode "D08". 
Chargeinformation and chargeprices is also seperated into flow for each entity.
Charge prices is now a an aggregation root and the architecture reflects this

- [x] Persist charge prices
- [x] ~~Make chargeprices an aggregat root~~
- [x] ~~Rename charges to chargeinformation and move to own folderstructure~~
- [x] Remove point from charges and chargelinks
- [x] Cleanup project
- [x] Write unittests for charge prices classes
- [x] Write integration test for charge prices
- [x] ~~Split chargeinformation and chargeprices into seperate test folders to reflect the domain~~
- [x] Merge with #1346

## References

* #1292

### CODE REVIEW

**Comment by Renetnielsen** [comment]
@cla-bot check

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/ChargeCommands/Handlers/ChargeIntegrationEventsPublisher.cs [hunk 1]
```diff
@@ -39,7 +40,7 @@ public async Task PublishAsync(ChargeCommandAcceptedEvent chargeCommandAcceptedE
 
             foreach (var chargeOperationDto in chargeCommandAcceptedEvent.Command.ChargeOperations)
             {
-                await _chargePublisher.PublishChargeCreatedAsync(chargeOperationDto).ConfigureAwait(false);
+                await _chargeInformationPublisher.PublishChargeInformationCreatedAsync(chargeOperationDto).ConfigureAwait(false);
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/ChargeCommands/Handlers/ChargeIntegrationEventsPublisher.cs:43]: Suggestion: keep the method name as it was, to reflect the [contract](https://github.com/Energinet-DataHub/geh-charges/blob/main/source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure/Contracts/Public/ChargeCreated/ChargeCreated.proto) it is publishing.
```suggestion
                await _chargeInformationPublisher.PublishChargeCreatedAsync(chargeOperationDto).ConfigureAwait(false);
```

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/ChargeInformations/Acknowledgement/ChargeInformationCreatedEvent.cs [hunk 2]
```diff
@@ -13,15 +13,15 @@
...
-using GreenEnergyHub.Charges.Domain.Charges;
+using GreenEnergyHub.Charges.Domain.Common;
 using GreenEnergyHub.Charges.Domain.Dtos.Messages;
 using NodaTime;
 
-namespace GreenEnergyHub.Charges.Application.Charges.Acknowledgement
+namespace GreenEnergyHub.Charges.Application.ChargeInformations.Acknowledgement
 {
-    public class ChargeCreatedEvent : IMessage
+    public class ChargeInformationCreatedEvent : IMessage
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/ChargeInformations/Acknowledgement/ChargeInformationCreatedEvent.cs:22]: I'm not sure we should do this renaming. It's related to an existing (and used) contract `ChargeCreated`.

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/ChargeInformations/Factories/ChargeInformationCreatedEventFactory.cs [hunk 3]
```diff
@@ -12,26 +12,26 @@
...
 using GreenEnergyHub.Charges.Core.Currency;
-using GreenEnergyHub.Charges.Domain.Charges;
+using GreenEnergyHub.Charges.Domain.ChargeInformations;
 using GreenEnergyHub.Charges.Domain.Dtos.ChargeCommands;
 
-namespace GreenEnergyHub.Charges.Application.Charges.Factories
+namespace GreenEnergyHub.Charges.Application.ChargeInformations.Factories
 {
-    public class ChargeCreatedEventFactory : IChargeCreatedEventFactory
+    public class ChargeInformationCreatedEventFactory : IChargeInformationCreatedEventFactory
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/ChargeInformations/Factories/ChargeInformationCreatedEventFactory.cs:22]: Related to previous comments; are we sure we wanna go with `ChargeInformation` here?

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/ChargeInformations/Factories/ChargeInformationCreatedEventFactory.cs [hunk 4]
```diff
@@ -12,26 +12,26 @@
...
             _currencyConfigurationIso4217 = currencyConfigurationIso4217;
         }
 
-        public ChargeCreatedEvent Create(ChargeOperationDto chargeOperationDto)
+        public ChargeInformationCreatedEvent Create(ChargeOperationDto chargeOperationDto)
         {
-            return new ChargeCreatedEvent(
-                chargeOperationDto.ChargeId,
+            return new ChargeInformationCreatedEvent(
+                chargeOperationDto.ChargeInformationId,
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/ChargeInformations/Factories/ChargeInformationCreatedEventFactory.cs:34]: I prefer to keep the prop name `ChargeId` on chargeOperationDto. 
Adding information to me seems confusing, when we are talking about the actual charge id.

```suggestion
                chargeOperationDto.ChargeInformationId,
```

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/ChargeInformations/Handlers/ChargeInformationHandler.cs [hunk 5]
```diff
@@ -153,26 +153,26 @@ public async Task HandleAsync(ChargeCommandReceivedEvent commandReceivedEvent)
...
+            var charge = await _chargeInformationFactory
                 .CreateFromChargeOperationDtoAsync(chargeOperationDto)
                 .ConfigureAwait(false);
 
-            await _chargeRepository.AddAsync(charge).ConfigureAwait(false);
+            await _chargeInformationRepository.AddAsync(charge).ConfigureAwait(false);
         }
 
-        private void HandleUpdateEvent(Charge charge, ChargeOperationDto chargeOperationDto)
+        private void HandleUpdateEvent(Domain.ChargeInformations.ChargeInformation chargeInformation, ChargeOperationDto chargeOperationDto)
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/ChargeInformations/Handlers/ChargeInformationHandler.cs:163]: Can `Domain.ChargeInformations.ChargeInformation`  be shortened?

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/ChargeInformations/Handlers/ChargeInformationHandler.cs [hunk 6]
```diff
@@ -190,13 +190,13 @@ private static OperationType GetOperationType(ChargeOperationDto chargeOperation
                 : OperationType.Update;
         }
 
-        private async Task<Charge?> GetChargeAsync(ChargeOperationDto chargeOperationDto)
+        private async Task<Domain.ChargeInformations.ChargeInformation?> GetChargeAsync(ChargeOperationDto chargeOperationDto)
         {
-            var chargeIdentifier = new ChargeIdentifier(
-                chargeOperationDto.ChargeId,
+            var chargeIdentifier = new ChargeInformationIdentifier(
+                chargeOperationDto.ChargeInformationId,
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/ChargeInformations/Handlers/ChargeInformationHandler.cs:196]: I'm not sure this should be named `ChargeInformationIdentifier`, I believe we will see the use of it in the price flow as well. To me, the information part of the name is misleading. I suggest we keep `new ChargeIdentifier`.

**Review by prtandrup — COMMENTED** [review]
A few comments so far. Still reviewing 😃

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/ChargeCommands/Handlers/ChargeIntegrationEventsPublisher.cs (hunk 1 above)
Comment (x-platformcoder) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/ChargeCommands/Handlers/ChargeIntegrationEventsPublisher.cs:43]: ok

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/ChargeInformations/Factories/ChargeInformationCreatedEventFactory.cs (hunk 3 above)
Comment (x-platformcoder) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/ChargeInformations/Factories/ChargeInformationCreatedEventFactory.cs:22]: ok

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/ChargeInformations/Acknowledgement/ChargeInformationCreatedEvent.cs (hunk 2 above)
Comment (x-platformcoder) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/ChargeInformations/Acknowledgement/ChargeInformationCreatedEvent.cs:22]: ok

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/ChargeInformations/Factories/ChargeInformationCreatedEventFactory.cs (hunk 4 above)
Comment (x-platformcoder) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/ChargeInformations/Factories/ChargeInformationCreatedEventFactory.cs]: ok

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/GreenEnergyHub.Charges.Application.csproj [hunk 7]
```diff
@@ -50,4 +50,8 @@ limitations under the License.
     </Compile>
   </ItemGroup>
 
+  <ItemGroup>
+    <Folder Include="ChargeCommands\Handlers\Message" />
+  </ItemGroup>
+
```
[the remark this answers is not in the document]
Comment (x-platformcoder) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/GreenEnergyHub.Charges.Application.csproj]: ok

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/ChargeInformations/Handlers/ChargeInformationHandler.cs (hunk 6 above)
Comment (x-platformcoder) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/ChargeInformations/Handlers/ChargeInformationHandler.cs]: ok

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/ChargePrices/Handlers/ChargePricesHandler.cs [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+using GreenEnergyHub.Charges.Domain.Dtos.Validation;
+
+namespace GreenEnergyHub.Charges.Application.ChargePrices.Handlers
+{
+    public class ChargePricesHandler : IChargePricesHandler
+    {
+        private readonly IChargeCommandReceiptService _chargeCommandReceiptService;
+        private readonly IInputValidator<ChargeOperationDto> _inputValidator;
+        private readonly IBusinessValidator<ChargeOperationDto> _businessValidator;
+        private readonly IChargeInformationRepository _chargeInformationInformationRepository;
```
[the remark this answers is not in the document]
Comment (x-platformcoder) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/ChargePrices/Handlers/ChargePricesHandler.cs]: fixed

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/ChargePrices/Handlers/ChargePricesHandler.cs [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+            _chargeCommandReceiptService = chargeCommandReceiptService;
+            _inputValidator = inputValidator;
+            _businessValidator = businessValidator;
+            _chargeInformationInformationRepository = chargeInformationInformationRepository;
+            _chargePriceRepository = chargePriceRepository;
+            _chargePriceFactory = chargePriceFactory;
+            _unitOfWork = unitOfWork;
+        }
+
+        public async Task HandleAsync(
```
[the remark this answers is not in the document]
Comment (x-platformcoder) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/ChargePrices/Handlers/ChargePricesHandler.cs:62]: I agree, but should we do this now or wait until we start implementing the validations for prices?

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/ChargeInformations/IChargeInformationRepository.cs [hunk 10]
```diff
@@ -16,21 +16,21 @@
...
 using System.Threading.Tasks;
 
-namespace GreenEnergyHub.Charges.Domain.Charges
+namespace GreenEnergyHub.Charges.Domain.ChargeInformations
 {
     /// <summary>
     /// Contract defining the capabilities of the infrastructure component facilitating interaction with the charges data store.
     /// </summary>
-    public interface IChargeRepository
+    public interface IChargeInformationRepository
```
[the remark this answers is not in the document]
Comment (x-platformcoder) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/ChargeInformations/IChargeInformationRepository.cs:24]: I think this must be decided by the team

**Pull request closed without merging**